dynamo_compile: Log all compilation time under all_compilation_types

c00w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149664

This counter is designed to include all compilation pytorch does (triton +
dynamo_compile). However this wasn't including all of dynamo compilation, since
it was put in at the fx_codegen_and_compile spot.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

c00w committed:
Update

[ghstack-poisoned]

c00w committed:
Update

[ghstack-poisoned]

masnesral commented on torch/_inductor/compile_fx.py:
```diff
@@ -626,6 +626,9 @@ def compile_fx_inner(
         # the counter here because we may dropped into compile_fx directly
         # from lazy backwards compilation.
         stack.enter_context(_WaitCounter("pytorch.wait_counter.dynamo_compile").guard())
+        stack.enter_context(
```
Do we not have exactly the same WaitCounter just named differently on the line above? Sorry I don't know what this is doing -- seems like they'd track the same thing

c00w replied:
This waitcounter includes triton autotuning etc... It's designed to be comprehensive of all compilation, not just torch.compile :).

masnesral replied:
Oh I see. This isn't a new waitcounter, you're just adding another location where it's incremented? If this is intended to be all compilation, then it doesn't seem complete? What about dynamo overhead?

masnesral replied:
I don't really get the comment above. This really seems like we're missing all dynamo overhead, but the comment is from Ed and I assume he knows more than I do.

c00w replied:
I agree that I want to include dynamo overhead, this is just incremental improvement.

re: dynamo measuring, yes today we're mostly missing it in this dataset.

c00w commented:
@pytorchbot merge

c00w committed:
Update on "dynamo_compile: Log all compilation time under all_compilation_types"


This counter is designed to include all compilation pytorch does (triton +
dynamo_compile). However this wasn't including all of dynamo compilation, since
it was put in at the fx_codegen_and_compile spot.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

c00w commented:
@pytorchbot merge

c00w committed:
Update

[ghstack-poisoned]

c00w commented:
@pytorchbot merge
